firebase-firestore/src/main/java/com/google/firebase/firestore/local/LocalStore.java

```diff
@@ -197,6 +209,11 @@ private void startMutationQueue() {
     localDocuments =
         new LocalDocumentsView(remoteDocuments, mutationQueue, documentOverlayCache, indexManager);
     queryEngine.setLocalDocumentsView(localDocuments);
+    queryEngine.setIndexManager(indexManager);
+
+    remoteDocuments.setIndexManager(indexManager);
```

Comment:
Do we need a test to check that the index manager is swapped when the user changes?

Reply:
We do - but I am not sure how to add them yet. The user change is only tested via Spec tests and we could do see the affect of switching the user once we also update the Query logic to use indexing by default. I added a TODO.